nodejs/eks/cni-addon.ts

```diff
@@ -384,133 +373,141 @@ export class VpcCniAddon extends pulumi.ComponentResource {
     }
 }
 
-function computeEnv(args: VpcCniAddonOptions): {
-    env: { [key: string]: string };
-    initEnv: { [key: string]: string };
-} {
-    const env: { name: string; value: string }[] = [];
-    const initEnv: { name: string; value: string }[] = [];
+export type ComputedEnv = {
+    env: ConfigurationValues;
+    initEnv: ConfigurationValues;
+};
+
+export function computeEnv(args: CniEnvVariables): ComputedEnv {
+    const env: { name: string; value: pulumi.Output<string> }[] = [];
+    const initEnv: { name: string; value: pulumi.Output<string> }[] = [];
     if (args.nodePortSupport) {
         env.push({
             name: "AWS_VPC_CNI_NODE_PORT_SUPPORT",
-            value: args.nodePortSupport ? "true" : "false",
+            value: pulumi.output(args.nodePortSupport).apply(stringifyBool),
```

Comment:
This is a super easy foot gun for component developers. You "just have to know" to do this.